pkg/test/env/factory_kafka.go

```diff
@@ -103,47 +103,55 @@ func (f *kafkaFactory) healthCheck() ComponentHealthCheck {
 	}
 }
 
-func (f *kafkaFactory) configureContainer(settings any) *containerConfig {
+func (f *kafkaFactory) configureContainer(settings any) *ContainerConfig {
 	s := settings.(*kafkaSettings)
 	hostName := "redpanda"
 
-	return &containerConfig{
+	return &ContainerConfig{
+		RunnerType: RunnerTypeLocal,
 		Hostname:   hostName,
 		Auth:       s.Image.Auth,
 		Repository: s.Image.Repository,
 		Tag:        s.Image.Tag,
-		PortBindings: portBindings{
-			"9092/tcp": s.BrokerPort,
-			"8081/tcp": s.SchemaRegistryPort,
+		PortBindings: PortBindings{
+			"main": {
+				ContainerPort: 9092,
+				HostPort:      s.BrokerPort,
+				Protocol:      "tcp",
+			},
+			"schema-registry": {
+				ContainerPort: 8081,
+				HostPort:      s.SchemaRegistryPort,
+				Protocol:      "tcp",
+			},
 		},
-		ExpireAfter: s.ExpireAfter,
 		Cmd: []string{
 			"redpanda start",
 			"--smp 1",
 			"--overprovisioned",
-			"--kafka-addr internal://0.0.0.0:19092,external://0.0.0.0:9092",
+			fmt.Sprintf("--kafka-addr internal://0.0.0.0:19092,external://0.0.0.0:%d", 9092),
```

Comment:
That change seems a little bit strange to me, why are we adding the second port, but not the first? Can we just keep the hardcoded string from before?